editor/scene/scene_paint_editor_plugin.h

```diff
@@ -0,0 +1,158 @@
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#pragma once
+
+#include "core/templates/hash_map.h"
+#include "editor/gui/editor_file_dialog.h"
+#include "editor/plugins/editor_plugin.h"
+#include "editor/scene/canvas_item_editor_plugin.h"
+#include "scene/2d/node_2d.h"
+#include "scene/gui/box_container.h"
+#include "scene/gui/check_box.h"
+#include "scene/gui/control.h"
+#include "scene/gui/item_list.h"
+#include "scene/gui/spin_box.h"
+#include "scene/gui/tree.h"
+
+class ScenePaintEditor : public Control {
+	GDCLASS(ScenePaintEditor, Control);
+
+	friend class ScenePaintEditorPlugin;
+
+	enum {
+		PICK_FILE_SYSTEM,
+		PICK_SCENE_TREE,
+		PICK_CANVAS_ITEM,
+	};
+
+	bool pinned = false;
+	bool scene_picker = false;
+	bool edit_properties = false;
+	bool grid = false;
+	bool snap_grid = false;
+	bool is_tool_selected = false;
+	bool is_painting = false;
+	bool is_erasing = false;
+
+	Size2i grid_step = Size2i(16, 16);
+	Vector2 last_paint_pos;
+	Rect2 paint_rect;
+	Vector2 mouse_pos;
+	Transform2D canvas_xform;
+
+	EditorUndoRedoManager *undo_redo = nullptr;
+	Popup *grid_settings_popup = nullptr;
+	CanvasItemEditor *canvas = nullptr;
+	Control *viewport = nullptr;
+	Node2D *node = nullptr;
+	Node2D *preview = nullptr;
+	Node2D *instance = nullptr;
+
+	HBoxContainer *toolbar = nullptr;
+
+	Button *pin_node_button = nullptr;
+	Button *scene_picker_button = nullptr;
+	Button *edit_properties_button = nullptr;
+
+	Button *grid_settings_button = nullptr;
+	CheckBox *grid_toggle_button = nullptr;
+	CheckBox *snap_grid_button = nullptr;
+	SpinBox *grid_step_x = nullptr;
+	SpinBox *grid_step_y = nullptr;
+
+	Node2D *selected_scene;
```

Comment:
```suggestion
	Node2D *selected_scene = nullptr;
```
Should fix the crash.